backend/endpoints/rom.py

```diff
@@ -299,8 +303,12 @@ def get_roms(
     )
 
     # Get the char index for the roms
-    char_index = db_rom_handler.get_char_index(query=query)
-    char_index_dict = {char: index for (char, index) in char_index}
+    char_index_dict = {}
+    if with_char_index:
+        char_index = db_rom_handler.with_char_index(
+            query=query, order_by_attr=order_by_attr
+        )
+        char_index_dict = {char: index for (char, index) in char_index}
```

Comment:
@zurdi15 for homepage queries we don't need to calculate the char_index, saving 2 complex DB queries